program/rust/build.rs

```diff
@@ -1,3 +1,21 @@
+mod build_utils;
+use bindgen;
+use std::collections::HashMap;
+use std::vec::Vec;
+
+
 fn main() {
     println!("cargo:rustc-link-search=../c/target");
+    
+    let borsh_derives: Vec<String> = vec!["BorshSerialize".to_string(), "BorshDeserialize".to_string()];
+
+    let parser = build_utils::DeriveAdderParserCallback{types_to_traits: HashMap::from([
+        ("cmd_hdr", borsh_derives)
+        //map more struct names to traits here, be sure that the definitions of your traites
+        //are included in c_oracle_header.rs
+    ])};
```

Comment:
It might be nice to put this in a function called `register_traits` or something similar, with a comment explaining why this is necessary.